android/src/main/java/com/mapbox/mapboxgl/MapboxMapController.java

```diff
@@ -290,18 +283,22 @@ public void setStyleString(String styleString) {
     }
   }
 
-  
+
 
   @SuppressWarnings({"MissingPermission"})
   private void enableLocationComponent(@NonNull Style style) {
     if (hasLocationPermission()) {
-      locationEngine = LocationEngineProvider.getBestLocationEngine(context);
+
       locationComponent = mapboxMap.getLocationComponent();
-      locationComponent.activateLocationComponent(
-          context, style, buildLocationComponentOptions(style));
+
+      LocationComponentActivationOptions options =
+              LocationComponentActivationOptions
+                      .builder(context, style)
+                      .locationComponentOptions(buildLocationComponentOptions(style))
+                      .build();
+
+      locationComponent.activateLocationComponent(options);
       locationComponent.setLocationComponentEnabled(true);
-      // locationComponent.setRenderMode(RenderMode.COMPASS); // remove or keep default?
-      locationComponent.setLocationEngine(locationEngine);
```

Comment:
The `locationComponent` has an engine by itself. It's working properly, and I tested it on a few different devices.